cypress/integration/upload-a-document.spec.ts

```diff
@@ -1,34 +1,86 @@
+import dsFixtures from '../fixtures/document-submission-response.json';
+
 describe('Can upload a document', () => {
+  const requestId = 'foo';
+
   beforeEach(() => {
     cy.intercept('/api/evidence/document_types', {
       fixture: 'document-types-response.json',
     });
-    cy.intercept('GET', '/api/evidence/evidence_requests/foo', {
+    cy.intercept('GET', `/api/evidence/evidence_requests/${requestId}`, {
       fixture: 'evidence-request-response-singular',
     });
-    cy.intercept('POST', '/api/evidence/document_submissions', {
-      fixture: 'document-submission-response-singular',
-      statusCode: 201,
-    });
+    cy.intercept('POST', '/api/evidence/document_submissions', (req) => {
+      const docType = req.body.documentType;
+      const response = dsFixtures.find((ds) => ds.documentType.id === docType);
+      req.reply((res) => {
+        res.send(201, response);
+      });
+    }).as('s3Upload');
 
-    cy.visit(`http://localhost:3000/resident/foo`);
+    cy.visit(`http://localhost:3000/resident/${requestId}`);
     cy.injectAxe();
   });
 
   it('Has no detectable accessibility issues', () => {
     cy.checkA11y();
   });
 
-  it('shows guidance and lets you upload a file', () => {
-    // View guidance
-    cy.get('h1').should('contain', 'You’ll need to photograph your documents');
-    cy.get('a').contains('Continue').click();
+  context('when upload is successful', () => {
+    beforeEach(() => {
+      cy.intercept('POST', dsFixtures[0].uploadPolicy.url, {
+        statusCode: 201,
+        delayMs: 2500,
+      });
+    });
+
+    it('shows guidance and lets you upload a file', () => {
+      // View guidance
+      cy.get('h1').should(
+        'contain',
+        'You’ll need to photograph your documents'
+      );
+      cy.get('a').contains('Continue').click();
 
-    // Attach a file
-    cy.get('h1').should('contain', 'Upload your documents');
-    cy.get('input[type=file]').attachFile('example.png');
-    cy.get('button').contains('Continue').click();
+      // Attach a file
+      cy.get('h1').should('contain', 'Upload your documents');
```

Comment:
Should the UI say 'document' instead of plural since we only support one document per submission?